lib/req_cookie_manager.ex

```diff
@@ -0,0 +1,85 @@
+if Code.ensure_loaded?(Req) do
+  defmodule HttpCookie.ReqCookieManager do
```

Comment:
You have this dependency for req:

```elixir
{:req, "~> 0.5.0", only: :test}
```

however for this to work it needs to be an optional dependency:

```elixir
{:req, "~> 0.5.0", optional: true}
```

(Keep that `~> 0.5.0`, I might have to remove :current_request_steps after all. I'll let you know.)

You can see this being broken here:

```elixir
iex> Mix.install [:req, {:http_cookie, github: "reisub/http_cookie", branch: "limits"}]
iex> Code.ensure_loaded?(HttpCookie.ReqCookieManager)
false
```

AFAIR for some time Mix prunes code paths between dependency compilations ensuring we don't have implicit dependencies and dependencies on the order of things.

Thanks for adding first class support for Req!